pkg/download/stasher/stasher.go

```diff
@@ -0,0 +1,115 @@
+package stasher
```

Comment:
Could we rename this pkg? Much of the difficulties I had reading this was due to this name. Like, it contains an impl. of a **protocol** which 'stashes' but the name is exactly as the `stash.Stasher` interface.

Reply:
@marpio very good point. Any idea what we should call it instead? 
I wanted to call it the "main" protocol, but obviously that name is taken lol.

Reply:
Yeah, naming is hard ;-)
But maybe something like this:
Separate the DP implementations into two categories
 - one that provides the core functionality which is to get the  required bits (mod, info, zip) from upstream. This would be i.e. implementation in the `download/goget` pkg or, in future, one that uses Olympus.
- and the wrappers/addons which provide some additional functionality like stash, worker pool, retry etc. So something like `protocol/addons` with functions like `func WithStash(p download.Protocol, ...)download.Protocol`

Reply:
@marpio i've seen this pattern with gRPC and I like it. But I still don't know what to name the "main" one that uses goget as an upstream and storage as a first stop and a stasher as an intermediary step :)

Reply:
@marpio I can have the addons package include the main one and just call it "WithStasher"?

Reply:
Yes, this sounds good!